docs/source/format/CDataInterface/PyCapsuleInterface.rst

```diff
@@ -0,0 +1,402 @@
+            requested_schema : PyCapsule | None
+                A PyCapsule containing a C ArrowSchema representation of a requested
+                schema. Conversion to this schema is best-effort.
+
+            Returns
+            -------
+            PyCapsule
+                A PyCapsule containing a C ArrowArrayStream representation of the
+                object.
+            """
+            ...
+
+Schema Requests
+---------------
+
+In some cases, there might be multiple possible Arrow representations of the
+same data. For example, a library might have a single integer type, but Arrow
+has multiple integer types with different sizes and sign. As another example,
+Arrow has several possible encodings for an array of strings: 32-bit offsets,
+64-bit offsets, string view, and dictionary-encoded. A sequence of strings could
+export to any one of these Arrow representations.
+
+In order to allow the caller to request a specific representation, the
+``__arrow_c_array__`` and ``__arrow_c_stream__`` methods may take an optional
+``requested_schema`` parameter. This parameter is a PyCapsule containing an
+``ArrowSchema``.
+
+The callee should attempt to provide the data in the requested schema. However,
+if the callee cannot provide the data in the requested schema, they may return
+with the same schema as if ``None`` were passed to ``requested_schema``.
+
+If the caller requests a schema that is not compatible with the data,
+say requesting a schema with a different number of fields, the callee should
+raise an exception. The requested schema mechanism is only meant to negotiate
+between different representations of the same data and not to allow arbitrary
+schema transformations.
+
+
+.. _PyCapsule: https://docs.python.org/3/c-api/capsule.html
+
+
+Examples
+========
+
+Create a PyCapsule
+------------------
+
+To create a PyCapsule, use the `PyCapsule_New <https://docs.python.org/3/c-api/capsule.html#c.PyCapsule_New>`_
+function. The function must be passed a destructor function that will be called
+to release the data the capsule points to. It must first call the release
+callback if it is not null, then free the struct.
+
+Below is the code to create a PyCapsule for an ``ArrowSchema``. The code for
+``ArrowArray`` and ``ArrowArrayStream`` is similar.
+
+.. tab-set::
+
+   .. tab-item:: C
+
+      .. code-block:: c
+
+         #include <Python.h>
+
+         void ReleaseArrowSchemaPyCapsule(PyObject* capsule) {
+             ArrowSchema* schema =
+                 (ArrowSchema*)PyCapsule_GetPointer(capsule, "arrow_schema");
+             if (schema->release != NULL) {
+                 schema->release(schema);
+             }
+             free(schema);
+         }
+         
+         PyObject* MakeArrowSchemaPyCapsule(ArrowSchema* schema) {
+           return PyCapsule_New(schema, "arrow_schema", ReleaseArrowSchemaPyCapsule);
+         }
+
+   .. tab-item:: Cython
+
+      .. code-block:: cython
+
+         import cpython
```

Comment:
Does this work or does this need to be `cimport cpython`?

Reply:
Yes, that needs to be `cimport`

Reply:
Thanks. I've now fixed the examples and tested they can compile successfully.